File: SparkWordCount/src/main/java/com/javadeveloperzone/spark/WordCount.java
package com.javadeveloperzone.spark;

import java.util.Arrays;
import java.util.Iterator;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.FlatMapFunction;
import org.apache.spark.api.java.function.Function2;
import org.apache.spark.api.java.function.PairFunction;

import scala.Tuple2;

public class WordCount {
	
	static class SplitFunction implements FlatMapFunction<String, String>
	{

		private static final long serialVersionUID = 1L;

		@Override
		public Iterator<String> call(String s) {
			return Arrays.asList(s.split(" ")).iterator();
		}
		
	}
	
	public static void main(String[] args)
	{

		SparkConf sparkConf = new SparkConf().setAppName("Simple Application");
		
		//Setting Master for running it from IDE.
		sparkConf.setMaster("local[2]");

		JavaSparkContext sc = new JavaSparkContext(sparkConf);

		JavaRDD<String> textFile = sc.textFile(args[0]);
		
		JavaRDD<String> words = textFile.flatMap(new SplitFunction());
		
		//
		JavaPairRDD<String, Integer> pairs = words
				.mapToPair(new PairFunction<String, String, Integer>() {
					public Tuple2<String, Integer> call(String s) {
						return new Tuple2<String, Integer>(s, 1);
					}
				});
		
		JavaPairRDD<String, Integer> counts = pairs.reduceByKey(new Function2<Integer, Integer, Integer>() {
					public Integer call(Integer a, Integer b) {
						return a + b;
					}
				});

		counts.saveAsTextFile(args[1]);
		sc.stop();
		sc.close();
	}
}

File: SparkWordCount/src/main/scala/com/javadeveloperzone/spark/scala/WordCount.scala
package com.javadeveloperzone.spark.scala

import org.apache.spark._
import org.apache.spark.SparkContext._

object WordCount {
  
   def main(args: Array[String]) 
   {
     
     /*By default we are setting local so it will run locally with one thread 
      *Specify: "local[2]" to run locally with 2 cores, OR 
      *        "spark://master:7077" to run on a Spark standalone cluster */
     
      val sparkContext = new SparkContext("local","Spark WordCount example using Scala",
          System.getenv("SPARK_HOME"))
      
      /*Reading input from File*/
      val input = sparkContext.textFile(args(0))
      
      /*Creating flat map from lines on input file*/
      val words = input.flatMap(line => line.split(" "))
      
      /*Performing mapping and reducing operation*/
      val counts = words.map(word => (word, 1)).reduceByKey{case (x,y) => x + y}
      
      /*Saving the result file to the location that we have specified as args[1]*/
      counts.saveAsTextFile(args(1))
      
    }
}